### ISSUE TITLE
Support Pytorch powered linear regression for downstream tasks.

### ISSUE THREAD

**Issue by zparis99** [leoniekerken/ECoG-foundation-model#10]
For simplicity the current encoding task is implemented using a sklearn linear regressor. This makes training really simple and easy to run on cpu's, but we could also consider handling it as a finetuning task for the model itself where we add a torch linear layer to the model output which will contain gradients and can be used to train the whole model after the VideoMAE training step.

This task would be to make a configurable option to using a torch linear layer and finetune the whole model as well for our downstream tasks.
